src/Agent.Worker/TaskManager.cs

```diff
@@ -83,6 +93,59 @@ into taskGrouping
                 }
             }
         }
+        public async Task RemoveNodeRunner(IExecutionContext executionContext)
+        {
+            var nodeHelper = new NodeHandlerHelper();
+
+            if (!nodeHelper.IsNodeFolderExist(NodeHandler.NodeFolder, HostContext))
+            {
+                executionContext.Debug($"Node 6 runner doesn't exist.");
+                return;
+            }
+
+            try
+            {
+                IOUtil.DeleteDirectory(Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), NodeHandler.NodeFolder), executionContext.CancellationToken);
+            }
+            catch (Exception ex)
+            {
+                Trace.Error($"Failed to delete Node 6 runner package folder. Exception: {ex}");
+            }
+        }
+        public async Task DownloadNodeRunnerAsync(IExecutionContext executionContext, IEnumerable<Pipelines.JobStep> steps)
+        {
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNull(steps, nameof(steps));
+
+            IEnumerable<Pipelines.TaskStep> tasks = steps.OfType<Pipelines.TaskStep>();
+
+            //Download Node 6 runner if needed
+            foreach (var task in tasks)
+            {
+                // Skip checkout task
+                if (task.Id == Pipelines.PipelineConstants.CheckoutTask.Id && task.Reference.Version == Pipelines.PipelineConstants.CheckoutTask.Version)
```

Comment:
I was thinking that we would not check if the pipeline needs node 6: that we'd just install it if node 6 is not enabled.  The advantage is that we can potentially install it earlier e.g. after the agent is registered, or during registration, so we don't delay executing the job